### DISCUSSION TITLE
Independent penalty to skills

### DISCUSSION THREAD

**Discussion by Odoamar** [fmasa/wfrp-master#104] [Ideas]
![Screenshot_2023-05-05-15-39-59-440_cz frantisekmasa dnd~2](https://user-images.githubusercontent.com/77120529/236476591-4e31370d-2699-4e70-9d23-70ece83197b5.jpg)
Managing skills looks great but sometimes constant penalties appeared for a character like:

- amputated fingers due critical hit
- wearing armor
- limited vision due to helmets
- off hand 
- any permanent condition

I would like to see:
1. Text box for Penaltry description
2. Counting tracker(I don't know correct component name) similar to this one above with negative values: from 99 to 0

**Comment by fmasa** [comment]
I'm not sure this is the right approach for several reasons:
- There may be penalties for various Skills or even Characteristics. The proposed approach would only work for Skills.
- This would be hard to use for non-permanent penalties (off-hand should be applied only when using a weapon off-hand, armour and helmet can be taken off).

I think this should be implemented as a separate concept (*Penalty*) with these fields:

- Name
- Description
- Affected Characteristics (zero or more)
- Affected skills (zero or more, ideally with wildcard support - `Language (Any)`)
- penalty value (0-99)
- Active: true/false

We may go even step further:

- Automatically apply penalties for Conditions
- Allow specifying penalties for Trappings and treat them as active when equipped/worn (not sure about that, I think this can be introduced later if useful because there are penalties that depend on different factors - i.e. opened/closed visor)
- In the future, this could be specified directly in Injuries, Mutations etc.

The advantage would be that we can show all current Penalties and let Players toggle Penalties that affect multiple Skills/Characteristics.

---

I don't know a good way to tackle Off-hand  right now as it's not a penalty per se, but rather Test-bound modifier. I would prefer to tackle this with other modifiers in https://github.com/fmasa/wfrp-master/issues/72

**Comment by Odoamar** [comment]
I like this approach. 

As you said, auto penalties could be problematic so toggle _Penalties_ are a great idea.
